src/main/java/org/redcrosswarriors/controllerservice/RequestBloodControllerService.java

```diff
@@ -0,0 +1,257 @@
+    {
+        boolean check = true;
+        try{
+            System.out.println("start");
+            requestRepository.newRequester(requestInput.getFirstName(),
+                    requestInput.getLastName(),
+                    requestInput.getEmail(),
+                    requestInput.getPhoneNumber(),
+                    requestInput.getBloodType(),
+                    requestInput.getHospitalName(),
+                    requestInput.getStreetName(),
+                    requestInput.getCityName(),
+                    requestInput.getStateName(),
+                    requestInput.getZipCode(),
+                    requestInput.getMessage());
+
+
+            System.out.println(requestInput.getLastName());
+            System.out.println(requestInput.getEmail());
+            System.out.println(requestInput.getPhoneNumber());
+            System.out.println(requestInput.getBloodType());
+            System.out.println(requestInput.getHospitalName());
+            System.out.println(requestInput.getStreetName());
+            System.out.println(requestInput.getCityName());
+            System.out.println(requestInput.getStateName());
+            System.out.println( requestInput.getZipCode());
+            System.out.println( requestInput.getMessage());
+
+
+
+            timeDetails = requestTimeRepository.getRequestedTimeByEmail(requestInput.getEmail());
+            String currentTime;
+            if(timeDetails == null)
+            {
+                currentTime = returnTime();
+                System.out.println(currentTime);
+                requestTimeRepository.requesterTimeUpdate(requestInput.getEmail(), currentTime);
+                List<String> matches;
+
+                matches = requestRepository.findMatches(requestInput.getBloodType());
+                if(matches.size() == 0)
+                {
+                    System.out.println("0 matches near by");
+                    //check = false;
+
+                }
+                else
+                {
+                    System.out.println("matches found: " + matches.size());
+                    SendNotificationEmail sendRequest = new SendNotificationEmail(matches, requestInput);
+                    sendRequest.start();
+                }
+
+            }
+            else
+            {
+                String lastTime;
+                lastTime = timeDetails.getTime();
+                currentTime = returnTime();
+                if(findDifference(lastTime, currentTime))
+                {
+                    requestTimeRepository.updateTime(currentTime, requestInput.getEmail());
+                    List<String> matches;
+                    matches = requestRepository.findMatches(requestInput.getBloodType());
+                    System.out.println("matches found: " + matches.size());
+                    SendNotificationEmail sendRequest = new SendNotificationEmail(matches, requestInput);
+                    sendRequest.start();
```

Comment:
Maybe this should me moved to it's own method since you do the same thing if the time is null. 
That way you can just do a method call in both if statements instead of having duplicate code